borg/archiver.py

```diff
@@ -177,6 +178,7 @@ def do_check(self, args, repository):
     @with_repository()
     def do_change_passphrase(self, args, repository, manifest, key):
         """Change repository key file passphrase"""
+        check_repository_compatibility(manifest, REPO_OPERATION_CHECK)
```

Comment:
true, the decorator is not always used. but it would cover at least the places where it is used.

additional in the existing decorator or just an additional call as you have now.